### PR TITLE
fix(external): #MC-285 fix external calendar flow

### PR DESCRIPTION
**Pull request by alicedraillard**
## Describe your changes
- can only create 1 calendar of each external platform
- front loader stops when eb from zimbra is received
- ical parsing fixed

## Checklist tests
for both platform and link external calendar:
- create calendar
- update calendar
then :
- sync calendars
- try to create 2nd Zimbra calendar

## Issue ticket number and link
[MC-285](https://jira.support-ent.fr/browse/MC-285)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by sjollois — APPROVED** [review]

File: src/main/java/net/atos/entng/calendar/models/CalendarModel.java [hunk 1]
```diff
@@ -25,18 +26,19 @@ public class CalendarModel {
...
-                .toInstant().toString();
+
+        Long updateDate = calendar.getJsonObject(Field.UPDATED, new JsonObject()).getLong(MongoField.$DATE, null);
+        this.updated = (updateDate != null) ? new Date(updateDate).toInstant().toString() :  null;
+
         this.color = calendar.getString(Field.COLOR, null);
         this.owner = calendar.getJsonObject(Field.OWNER, new JsonObject());
         this.isExternal = calendar.getBoolean(Field.ISEXTERNAL, false);
-        this.icsLink = calendar.getString(Field.ICSLINK);
+        this.icsLink = calendar.getString(Field.ICSLINK, null);
```
Comment (qpic-cgi) [inline src/main/java/net/atos/entng/calendar/models/CalendarModel.java:36]: pas utile

**Review by qpic-cgi — APPROVED** [review]

File: build.gradle [hunk 2]
```diff
@@ -59,7 +59,7 @@ dependencies {
   compileOnly "io.vertx:vertx-core:$vertxVersion"
   compile "io.vertx:vertx-web-client:$vertxVersion"
   compile "org.entcore:common:$entCoreVersion"
-  compile "org.mnode.ical4j:ical4j:2.0.2"
+  compile "org.mnode.ical4j:ical4j:3.2.10"
```
Comment (alicedraillard) [inline build.gradle:62]: update ical4j version

File: src/main/java/net/atos/entng/calendar/Calendar.java [hunk 3]
```diff
@@ -72,6 +72,7 @@ public void start() throws Exception {
         vertx.deployVerticle(ICalHandler.class.getName(), new DeploymentOptions().setWorker(true).setConfig(config));
 
         setRepositoryEvents(new CalendarRepositoryEvents(vertx));
+        System.setProperty("ical4j.unfolding.relaxed", "true");
```
Comment (alicedraillard) [inline src/main/java/net/atos/entng/calendar/Calendar.java:75]: handle parsing errors for ical

File: src/main/java/net/atos/entng/calendar/helpers/CalendarHelper.java [hunk 4]
```diff
@@ -110,10 +108,14 @@ private Future<Void> prepareCalendarAndEventsForUpdate(JsonObject params, JsonAr
 
                 updateExternalCalendar(params, calendar, true)
                         .compose(result -> {
-                            Date date = new Date(calendar.getJsonObject(Field.UPDATED).getLong(MongoField.$DATE));
-                            SimpleDateFormat simpleDateFormat = new SimpleDateFormat(DateUtils.DATE_FORMAT_UTC);
-                            simpleDateFormat.setTimeZone(TimeZone.getTimeZone(ZoneId.of("UTC")));
-                            return eventServiceMongo.deleteDatesAfterComparisonDate(calendar.getString(Field._ID), simpleDateFormat.format(date));
+                            if (calendar.getJsonObject(Field.UPDATED, null) == null) {
```
Comment (alicedraillard) [inline src/main/java/net/atos/entng/calendar/helpers/CalendarHelper.java:111]: HERE

File: src/main/java/net/atos/entng/calendar/controllers/CalendarController.java [hunk 5]
```diff
@@ -473,17 +473,24 @@ public void calendarEventBusHandler(Message<JsonObject> message) {
                     JsonObject params = new JsonObject();
                     String local = null;
                     try {
-                        local = new JsonObject((String) ((LinkedHashMap<?, ?>) user.getAttribute(Field.PREFERENCES)).get(Field.LANGUAGE)).getString(Field.DEFAULT_DOMAIN);
+                        String systemDomainLanguage = (String) ((LinkedHashMap<?, ?>) user.getAttribute(Field.PREFERENCES)).get(Field.LANGUAGE);
+                        String systemDomain = systemDomainLanguage != null ? new JsonObject(systemDomainLanguage).getString(Field.DEFAULT_DOMAIN, null) : null;
+                        local = systemDomain != null ? systemDomain : "fr";
                     } catch (Exception ignored) {
+                        String errMessage = String.format("[Calendar@%s::calendarEventBusHandler]: case 'zimbra-platform-ics': " +
```
Comment (alicedraillard) [inline src/main/java/net/atos/entng/calendar/controllers/CalendarController.java:480]: HERE

**Review by sjollois — APPROVED** [review]

**Review by aymericdevriesere — APPROVED** [review]

**Pull request merged**